crates/ty_python_semantic/src/types/narrow.rs

```diff
@@ -1636,3 +1743,32 @@ fn all_matching_typeddict_fields_have_literal_types<'db>(
         _ => true,
     }
 }
+
+fn is_supported_tuple_tag_literal(ty: Type) -> bool {
+    matches!(
+        ty,
+        // Same as TypedDict: support string, bytes, and int literals.
+        // TODO: We'd like to support `EnumLiteral` also, but we have to be careful with types like
+        // `IntEnum` and `StrEnum` that have custom `__eq__` methods.
+        Type::StringLiteral(_) | Type::BytesLiteral(_) | Type::IntLiteral(_)
+    )
+}
```

Comment:
can we not just rename `is_supported_typeddict_tag_literal` to `is_supported_tag_literal` and use that in both places?